src/sec_certs/dataset/cc.py

```diff
@@ -780,164 +757,38 @@ def _compute_cert_labs(self) -> None:
-
-    def _compute_heuristics(self) -> None:
-        self._compute_normalized_cert_ids()
-        super()._compute_heuristics()
-        self._compute_scheme_data()
-        self._compute_cert_labs()
-        self._compute_sars()
-
-    @staged(logger, "Computing heuristics: references between certificates.")
-    def _compute_references(self) -> None:
-        def ref_lookup(kw_attr):
-            def func(cert):
-                kws = getattr(cert.pdf_data, kw_attr)
-                if not kws:
-                    return set()
-                res = set()
-                for scheme, matches in kws["cc_cert_id"].items():
-                    for match in matches:
-                        try:
-                            canonical = CertificateId(scheme, match).canonical
-                            res.add(canonical)
-                        except Exception:
-                            res.add(match)
-                return res
-
-            return func
-
-        for ref_source in ("report", "st"):
-            kw_source = f"{ref_source}_keywords"
-            dep_attr = f"{ref_source}_references"
-
-            finder = ReferenceFinder()
-            finder.fit(self.certs, lambda cert: cert.heuristics.cert_id, ref_lookup(kw_source))  # type: ignore
-
-            for dgst in self.certs:
-                setattr(self.certs[dgst].heuristics, dep_attr, finder.predict_single_cert(dgst, keep_unknowns=False))
-
-    @serialize
-    def process_auxiliary_datasets(self, download_fresh: bool = False) -> None:
-        """
-        Processes all auxiliary datasets needed during computation. On top of base-class processing,
-        CC handles protection profiles, maintenance updates and schemes.
-        """
-        super().process_auxiliary_datasets(download_fresh)
-        self.auxiliary_datasets.pp_dset = self.process_protection_profiles(to_download=download_fresh)
-        self.auxiliary_datasets.mu_dset = self.process_maintenance_updates(to_download=download_fresh)
-        self.auxiliary_datasets.scheme_dset = self.process_schemes(
-            to_download=download_fresh, only_schemes={cert.scheme for cert in self}
+        for scheme in self.aux_handlers[CCSchemeDatasetHandler].dset:
+            if certified := scheme.lists.get(EntryType.Certified):
+                certs = [cert for cert in self if cert.status == "active"]
+                matches = CCSchemeMatcher.match_all(certified, scheme.country, certs)
+                for dgst, match in matches.items():
+                    self[dgst].heuristics.scheme_data = match
+            if archived := scheme.lists.get(EntryType.Archived):
+                certs = [cert for cert in self if cert.status == "archived"]
+                matches = CCSchemeMatcher.match_all(archived, scheme.country, certs)
+                for dgst, match in matches.items():
+                    self[dgst].heuristics.scheme_data = match
```

Comment:
Isn't this duplicate? I saw something like this function in the `heuristics` subpackage.

Reply:
Fixed.